packages/medusa/src/repositories/money-amount.ts

```diff
@@ -43,6 +48,42 @@ export class MoneyAmountRepository extends Repository<MoneyAmount> {
     return pricesNotInPricesPayload
   }
 
+  public async deleteVariantPricesNotIn(
+    variantId: string,
+    prices: Price[]
+  ): Promise<void> {
+    const existingPricesIdsQuery = this.createQueryBuilder()
+      .select("id")
+      .where({
+        variant_id: variantId,
+        price_list_id: IsNull(),
+      })
+      .andWhere(
+        new Brackets((qb) => {
+          prices.forEach((price) => {
+            qb.orWhere(
+              new Brackets((qb) => {
+                qb.where({
+                  currency_code: price.currency_code,
+                  region_id: IsNull(),
+                }).orWhere({ region_id: price.region_id })
+              })
+            )
+          })
+        })
+      )
+
+    await this.createQueryBuilder()
```

Comment:
Feel free to push a commit to the branch with it, I have not been able to write a delete query that typeorm accepted and where the outcome was the same as this. So if you know how to, it would be awesome 😅

Reply:
I add it to my today todo if that's fine with you 💪

Reply:
Sounds good, but depending on when we are planning on releasing I think we should release this as is if you don't get the time (which is totally fair!), as it fixes a quite significant bug, even if it is a little less performant than what it could be.

Reply:
yes completely agree. Let's do that and I ll do a separate PR for that part 💪

Reply:
@kasperkristensen could you have a look to [that](https://github.com/medusajs/medusa/pull/3152/commits/8e40c0b2b5bbd3e96dd012f1a9f54ff384bd55fe) please :D

Reply:
cc @olivermrbl